deviceModeInit.js

```diff
@@ -467,6 +467,34 @@ var rudderTracking = (function () {
       });
   }
 
+  /**
+   * Checks whether the url for the product contains the variant id
+   * @returns {string} variant id or null
+   */
+  function findVariantIdInURL() {
+    const matches = window.location.href.match(/\d{8,20}/);
+    if(matches) {
+      return matches[0]
+    }
+    return null;
+  };
+
+  /**
+   * Returns the sku value for the variant matching the id in url 
+   * @param {*} payload 
+   * @returns {string} matching variant's sku or undefined
+   */
+  function getVariantSku(payload) {
+    const variantId = findVariantIdInURL();
+    const variant = { payload }
```

Comment:
Put a check if variandId is available or not, otherwise we don't need to run the for loop.

Reply:
Done.